### PR TITLE
Migrate to AWS sdk v2: S3 client

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #1663 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/filestore/s3/s3.go [hunk 1]
```diff
@@ -84,55 +79,47 @@ func NewStore(region, bucket string, opts ...Option) (*Store, error) {
...
-			case s3.ErrCodeInvalidObjectState:
-				err = fmt.Errorf("invalid object state: %w", err)
-				return
-			default:
-				err = fmt.Errorf("unexpected aws error given: %w", err)
-				return
-			}
-		}
-		err = fmt.Errorf("unknown error given: %w", err)
-		return
```
Comment (nghialv) [inline pkg/filestore/s3/s3.go:135]: Why was this part removed?

Comment (khanhtc1202) [reply pkg/filestore/s3/s3.go:135]: Nice catch 👍 I missed handling in case not found s3 object 🙏

Comment (khanhtc1202) [reply pkg/filestore/s3/s3.go:135]: addressed on https://github.com/pipe-cd/pipe/pull/1671/commits/5a02024e8edf63536d6edb302993b4d8495f694a 🙏

**Comment by nghialv** [comment]
Nice. More readable code with AWS SDK v2 (compared to v1).
/approve

**Comment by khanhtc1202** [comment]
Glad to hear that 😁 Thank you 🙏
